platform/ui/src/contextProviders/HangingProtocolProvider.jsx

```diff
@@ -0,0 +1,140 @@
+import React, {
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useReducer,
+} from 'react';
+import PropTypes from 'prop-types';
+
+const DEFAULT_STATE = {
+  hangingProtocol: null,
+  hpAlreadyApplied: {}
+};
+
+export const HangingProtocolContext = createContext(DEFAULT_STATE);
+
+export function HangingProtocolProvider({ children, service }) {
+  const hangingProtocolReducer = (state, action) => {
+    switch (action.type) {
+      case 'SET_HANGING_PROTOCOL': {
+        return { ...state, ...{ hangingProtocol: action.payload.hangingProtocol } };
+      }
+      case 'SET_HANGING_PROTOCOL_ALREADY_APPLIED': {
```

Comment:
It was unclear what this does until you look through the code.

> SET_VIEWPORTINDEX_APPLIED

Or something.
- Per viewport.